src/services/acspDataService.ts

```diff
@@ -10,12 +10,16 @@ export class AcspDataService {
         const userId = session?.data?.signin_info?.user_profile?.id!;
         try {
             if (acspData === undefined) {
+                const { v4: uuidv4 } = require("uuid");
+                const applicationId = uuidv4();
+
                 acspData = {
-                    id: userId,
+                    id: applicationId,
```

Comment:
Why are we creating the id, can we not use Mongo's capability to create id for us.  Otherwise there is an edge case of hitting duplicate id issue.

Reply:
resolved